relayer/naive-strategy.go

```diff
@@ -232,12 +237,68 @@ func (nrs *NaiveStrategy) UnrelayedAcknowledgements(src, dst *Chain) (*RelaySequ
 
 // HandleEvents defines how the relayer will handle block and transaction events as they are emitted
 func (nrs *NaiveStrategy) HandleEvents(src, dst *Chain, events map[string][]string) {
+	// check for misbehaviour and submit if found
+	err := checkAndSubmitMisbehaviour(src, events)
+	if err != nil {
+		src.Error(err)
+	}
+
 	rlyPackets, err := relayPacketsFromEventListener(src.PathEnd, dst.PathEnd, events)
 	if len(rlyPackets) > 0 && err == nil {
 		nrs.sendTxFromEventPackets(src, dst, rlyPackets)
 	}
 }
 
+func checkAndSubmitMisbehaviour(src *Chain, events map[string][]string) error {
+	if hdrs, ok := events[fmt.Sprintf("%s.%s", updateCliTag, headerTag)]; ok {
+		for i, hdr := range hdrs {
+			clientIDs := events[fmt.Sprintf("%s.%s", updateCliTag, clientIDTag)]
+			emittedClientID := clientIDs[i]
+
+			if src.PathEnd.ClientID == emittedClientID {
+				hdrBytes, err := hex.DecodeString(hdr)
+				if err != nil {
+					return err
+				}
+
+				exportedHeader, err := clienttypes.UnmarshalHeader(src.Encoding.Marshaler, hdrBytes)
+				if err != nil {
+					return err
+				}
+
+				emittedHeader, ok := exportedHeader.(*tmclient.Header)
+				if !ok {
+					return fmt.Errorf("emitted header is not tendermint type")
+				}
+
+				trustedHeader, err := src.GetLightSignedHeaderAtHeight(emittedHeader.Header.Height)
```

Comment:
So, this function currently returns a header without a trusted validator set and trusted height. We need to inject these into the header. 

This gets a little complicated. It is possible the trusted header used to do a valid update of `emittedHeader` is also misbehavior. We need to iterate through all consensus states and check that all the previous submitted updates are valid. If this occurs we need to parse the tx that created the past misbehaviour and construct misbehaviour against that.

Lets handle this case in a different pr. For now, take the trusted validator set and height of the emitted header and set those value for the trusted header

```go
trustedHeader.TrustedValidators = emittedHeader.TrustedValidators
trustedHeader.TrustedHeight = emittedHeader.TrustedHeight
```